### PR TITLE
Fix tests and document cleanup

### PR DESCRIPTION
**Pull request by hsiang9431-amzn**
##  opensearch-project/security-dashboards-plugin pull request intake form
_Please provide as much details as possible to get feedback/acceptance on your PR quickly_

 
 1. __Category:__ (_Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation_)
 Enhancement, Test fix
 
 
 2. __Github Issue # or road-map entry, if available:__
N/A

 3. __Description of changes:__
- Fix Integration tests
- Integration tests now use docker image built by infra, no longer downloads its own dependencies
- Minor fixes for Unit tests
- Update Readme to display badges for both unit and integration tests
- Remove unused test helper files

 4. __Why these changes are required?__ 

 5. __What is the old behavior before changes and new behavior after changes?__ (_Please add any example/logs/screen-shot if available_)

 6. __Testing done:__ (_Please provide details of testing done: Unit testing, integration testing and manual testing_)
 Tested with GitHub workflow
 

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/integration-test.yml [hunk 1]
```diff
@@ -13,81 +13,83 @@ jobs:
...
-          timeout 900 bash -c 'while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' https://localhost:9200 -u admin:admin -k)" != "200" ]]; do echo "ping ElasticSearch..."; sleep 5; done'
+          echo "::set-output name=opendistro_version::$(jq -r '.version' ./OpenSearch-Dashboards/plugins/security-dashboards-plugin/package.json | sed 's/[0-9]*$/0/')"
       
-      - name: Use kibana release commit
+      - name: Use OpenSearch Dashboards release commit
         run: |
-          cd ./kibana
-          git checkout tags/v${{ steps.kbn_version.outputs.kbn_version }} -b v${{ steps.kbn_version.outputs.kbn_version }}
+          cd ./OpenSearch-Dashboards
+          # git checkout tags/v${{ steps.osd_version.outputs.osd_version }} -b v${{ steps.osd_version.outputs.osd_version }}
```
Comment (cliu123) [inline .github/workflows/integration-test.yml:61]: why commented out?

File: .github/workflows/integration-test.yml [hunk 2]
```diff
@@ -13,81 +13,83 @@ jobs:
...
+
+      - name: Bootstrap OpenSearch Dashboards
         run: |
-          cd ./kibana
-          yarn kbn bootstrap --oss
+          cd ./OpenSearch-Dashboards
+          yarn osd bootstrap
       
-      - name: Run integration tests
+      - name: Run integration test
```
Comment (cliu123) [inline .github/workflows/integration-test.yml:90]: If there are more than 1 tests, should we retain `tests`

File: .github/workflows/unit-test.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-name: CI
```
Comment (cliu123) [inline .github/workflows/unit-test.yml:1]: Why CI gets renamed?

File: .github/workflows/integration-test.yml (hunk 2 above)
Comment (hsiang9431-amzn) [reply .github/workflows/integration-test.yml]: hmm..i thought I put tests here
fixed with latest push

File: .github/workflows/integration-test.yml (hunk 1 above)
Comment (hsiang9431-amzn) [reply .github/workflows/integration-test.yml:61]: this is for getting latest commit on dashboards. they may be adding changes that aren't captured by tags

File: .github/workflows/unit-test.yml (hunk 3 above)
Comment (hsiang9431-amzn) [reply .github/workflows/unit-test.yml:1]: that job is actually unit tests. CI is too broad for that

File: .github/workflows/integration-test.yml (hunk 1 above)
Comment (cliu123) [reply .github/workflows/integration-test.yml:61]: The latest commit is actually not as stable as tags. Tags got cut when some bugs are found and fixed. So tags are actually considered as a stable version, which is why tags are provided for Integ Tests and Unit Tests for plugins. This the case on OpenSearch. Is Dashboards different?

File: test/es/setup_es.js [hunk 4]
```diff
@@ -1,20 +0,0 @@
...
- *   You may not use this file except in compliance with the License.
- *   A copy of the License is located at
- *
- *       http://www.apache.org/licenses/LICENSE-2.0
- *
- *   or in the "license" file accompanying this file. This file is distributed
- *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
- *   express or implied. See the License for the specific language governing
- *   permissions and limitations under the License.
- */
```
Comment (cliu123) [inline test/es/setup_es.js:14]: @saratvemulapalli Please review this. Is it ok to remove the license headers?

Comment (cliu123) [reply test/es/setup_es.js:14]: NVM. The entire file got removed.

File: .github/workflows/integration-test.yml [hunk 5]
```diff
@@ -13,81 +13,78 @@ jobs:
...
-          timeout 900 bash -c 'while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' https://localhost:9200 -u admin:admin -k)" != "200" ]]; do echo "ping ElasticSearch..."; sleep 5; done'
+          echo "::set-output name=osd_version::$(jq -r '.opensearchDashboards.version' ./OpenSearch-Dashboards/plugins/security-dashboards-plugin/package.json)"
       
-      - name: Use kibana release commit
+      - name: Check OpenSearch Dashboards release tag
         run: |
-          cd ./kibana
-          git checkout tags/v${{ steps.kbn_version.outputs.kbn_version }} -b v${{ steps.kbn_version.outputs.kbn_version }}
+          cd ./OpenSearch-Dashboards
+          git checkout tags/${{ steps.osd_version.outputs.osd_version }} -b v${{ steps.osd_version.outputs.osd_version }}
```
Comment (cliu123) [inline .github/workflows/integration-test.yml:56]: Why removing the `v`?

**Review by cliu123 — APPROVED** [review]

**Pull request merged**